pkg/controller/injector/k8s_binding_injector.go

```diff
@@ -64,15 +63,10 @@ func (b *k8sBindingInjector) Inject(binding *servicecatalog.Binding, cred *broke
 		Data: make(map[string][]byte),
 	}
 
-	// For each item in the cred just serialize its value into JSON and
-	// save it in the secret
+	// For each item in the cred, just convert it to the bytes for a string
+	// representation of the value
 	for k, v := range *cred {
-		data, err := json.Marshal(v)
-		if err != nil {
-			return fmt.Errorf("Unable to marshal credential value %q: %s",
-				k, err)
-		}
-		secret.Data[k] = data
+		secret.Data[k] = []byte(fmt.Sprintf("%v", v))
```

Comment:
Ugh. Ya. I'd somehow talked myself into believing creds couldn't _possibly_ be arbitrarily nested JSON structures like that, but after double-checking the OSB spec, they can be. fml